### PR TITLE
feat: add Reality short ID rotation

### LINKED ISSUE #107: feat: Auto-rotate Reality Short ID on schedule
## Background

Reality protocol uses a Short ID as part of its handshake. Periodically rotating this value can reduce the window for traffic analysis and fingerprinting.

## Motivation

- Enhances operational security by limiting the lifespan of any single Short ID
- Reduces risk if a Short ID is leaked or observed
- Can be implemented as a simple cron job or systemd timer
- Minimal disruption if clients are notified or auto-updated via subscription endpoint

## Requirements

- [ ] Add `sbx rotate-shortid` command to generate and apply a new Short ID
- [ ] Add optional `--schedule` flag to set up automatic rotation (e.g., `--schedule weekly`)
- [ ] Implement as a systemd timer or cron job
- [ ] Automatically update client configs / subscription endpoint after rotation
- [ ] Log rotation events for audit trail
- [ ] Add `--dry-run` flag to preview changes without applying
- [ ] Persist rotation history in `state.json`
- [ ] Add unit tests

## Dependencies

- Issue #101 (Subscription endpoint) — for automatic client config updates after rotation

**Priority:** Low

### PR DESCRIPTION
**Pull request by xrf9268-hue**
## Summary
- add `sbx rotate-shortid` with `--dry-run`, `--schedule`, timer-driven execution, audit history, and install/uninstall integration for Reality short ID rotation
- harden runtime mutation safety by serializing rotation with the shared mutation lock, using staged multi-file restore paths, and covering the lock/rollback edge cases with regression tests
- record the lock-order and staged-restore lessons in `.claude/CODING_STANDARDS.md` so future runtime mutators follow the same discipline

## Test Plan
- `bash tests/test-runner.sh unit`
- `bash -u install.sh --help`
- on the AWS VM: `bash tests/unit/test_reality_rotation.sh`
- on the AWS VM: `sudo env AUTO_INSTALL=1 bash install.sh`
- on the AWS VM: `sudo sbx rotate-shortid --dry-run`
- on the AWS VM: `sudo sbx rotate-shortid`
- on the AWS VM: `sudo sbx rotate-shortid --schedule weekly`
- on the AWS VM: `sudo systemctl start sbx-shortid-rotate.service`
- on the AWS VM: `sudo sbx rotate-shortid --schedule off`
- on the AWS VM: `bash scripts/e2e/install-lifecycle-smoke.sh`

Closes #107.

### CODE REVIEW

File: lib/reality_rotation.sh [hunk 1]
```diff
@@ -0,0 +1,802 @@
...
+  local timer_unit_content=''
+
+  service_unit_path=$(_rotation_service_unit_path)
+  timer_unit_path=$(_rotation_timer_unit_path)
+  service_unit_content=$(_rotation_service_unit_content)
+  timer_unit_content=$(_rotation_timer_unit_content "${on_calendar}")
+
+  install_systemd_unit "${service_unit_path}" "${service_unit_content}"
+  install_systemd_unit "${timer_unit_path}" "${timer_unit_content}"
+  systemctl enable --now "${ROTATION_TIMER_NAME}" >/dev/null 2>&1
```
[the remark this answers is not in the document]
Comment (xrf9268-hue) [reply lib/reality_rotation.sh]: Fixed in `e988c9c`.

`_rotation_install_units()` now fails fast on each unit write / enable step, and `install_systemd_unit()` no longer returns success after failed writes or `daemon-reload`. I also added a regression test that removes the target systemd directory and verifies `reality_rotation_schedule weekly` fails without persisting `enabled=true` into `state.json`.
